src/main/java/org/apache/pirk/responder/wideskies/spark/streaming/ComputeStreamingResponse.java

```diff
@@ -0,0 +1,467 @@
+ * 
+ */
+public class ComputeStreamingResponse
+{
+  private static final Logger logger = LoggerFactory.getLogger(ComputeStreamingResponse.class);
+
+  private String dataInputFormat = null;
+  private String inputData = null;
+  private String outputFile = null;
+  private String outputDirExp = null;
+
+  private String queryInput = null;
+  QuerySchema qSchema = null;
+
+  private String esQuery = "none";
+  private String esResource = "none";
+
+  private FileSystem fs = null;
+  private HadoopFileSystemStore storage = null;
+  private JavaStreamingContext jssc = null;
+
+  boolean useQueueStream = false;
+
+  private long batchSeconds = 0;
+  private long windowLength = 0;
+
+  private Accumulators accum = null;
+  private BroadcastVars bVars = null;
+
+  private QueryInfo queryInfo = null;
+  Query query = null;
+
+  private int numDataPartitions = 0;
+  private int numColMultPartitions = 0;
+
+  private boolean colMultReduceByKey = false;
+
+  public ComputeStreamingResponse(FileSystem fileSys) throws Exception
+  {
+    fs = fileSys;
+    storage = new HadoopFileSystemStore(fs);
+
+    dataInputFormat = SystemConfiguration.getProperty("pir.dataInputFormat");
+    if (!InputFormatConst.ALLOWED_FORMATS.contains(dataInputFormat))
+    {
+      throw new IllegalArgumentException("inputFormat = " + dataInputFormat + " is of an unknown form");
+    }
+    logger.info("inputFormat = " + dataInputFormat);
+    if (dataInputFormat.equals(InputFormatConst.BASE_FORMAT))
+    {
+      inputData = SystemConfiguration.getProperty("pir.inputData", "none");
+      if (inputData.equals("none"))
+      {
+        throw new IllegalArgumentException("For inputFormat = " + dataInputFormat + " an inputFile must be specified");
+      }
+      logger.info("inputFile = " + inputData);
+    }
+    else if (dataInputFormat.equals(InputFormatConst.ES))
+    {
+      esQuery = SystemConfiguration.getProperty("pir.esQuery", "none");
+      esResource = SystemConfiguration.getProperty("pir.esResource", "none");
+      if (esQuery.equals("none"))
+      {
+        throw new IllegalArgumentException("esQuery must be specified");
+      }
+      if (esResource.equals("none"))
+      {
+        throw new IllegalArgumentException("esResource must be specified");
+      }
+      logger.info("esQuery = " + esQuery + " esResource = " + esResource);
+    }
+    outputFile = SystemConfiguration.getProperty("pir.outputFile");
+    outputDirExp = outputFile + "_exp";
+
+    queryInput = SystemConfiguration.getProperty("pir.queryInput");
+    String stopListFile = SystemConfiguration.getProperty("pir.stopListFile");
+
+    logger.info("outputFile = " + outputFile + " queryInputDir = " + queryInput + " stopListFile = " + stopListFile + " esQuery = " + esQuery
+        + " esResource = " + esResource);
+
+    // Pull the batchSeconds and windowLength parameters
+    batchSeconds = Long.parseLong(SystemConfiguration.getProperty("pir.sparkstreaming.batchSeconds", "30"));
```

Comment:
'batchSeconds' is an input to the Durations class' static seconds method which requires a long (http://spark.apache.org/docs/latest/api/java/) 

But, yes, I do think that a long is a bit of overkill ;)